dbt_platform_helper/providers/config.py

```diff
@@ -19,23 +16,7 @@ def __init__(self, config_validator, echo=click.secho):
         self.config = {}
         self.validator = config_validator
         self.echo = echo
-
-    # TODO - this is to do with Yaml validation
-    def lint_yaml_for_duplicate_keys(self, file_path: str, lint_config=None):
-        if lint_config is None:
-            lint_config = {"rules": {"key-duplicates": "enable"}}
-
-        with open(file_path, "r") as yaml_file:
-            file_contents = yaml_file.read()
-            results = linter.run(file_contents, YamlLintConfig(yaml.dump(lint_config)))
-
-        parsed_results = [
-            "\t"
-            + f"Line {result.line}: {result.message}".replace(" in mapping (key-duplicates)", "")
-            for result in results
-        ]
-
-        return parsed_results
+        self.yaml_file_provider = YamlFileProvider
```

Comment:
for test config provider should we inject this via the parameters. Also would making the attribute make more sense to just be `file_provider`